src/sfCommand.ts

```diff
@@ -31,14 +33,22 @@ export abstract class SfCommand<T> extends Command {
   public static configurationVariablesSection?: HelpSection;
   public static envVariablesSection?: HelpSection;
   public static errorCodes?: HelpSection;
+  public static tableFlags = ux.table.flags;
 
   public spinner: Spinner;
+  public progress: Progress;
 
   private warnings: SfCommand.Warning[] = [];
+  private ux: Ux;
+  private prompter: Prompter;
 
   public constructor(argv: string[], config: Config) {
     super(argv, config);
-    this.spinner = new Spinner(this.jsonEnabled());
+    const outputEnabled = !this.jsonEnabled();
+    this.spinner = new Spinner(outputEnabled);
+    this.progress = new Progress(outputEnabled);
+    this.ux = new Ux(outputEnabled);
+    this.prompter = new Prompter();
```

Comment:
Not all commands will need these, I'd prefer to have the individual command instantiate them if it needs to.

Reply:
@RodEsp Can you elaborate on why that's your preference?

Reply:
Basically for performance. It might be minimal but there's no need to use up memory for commands that will never use one or more of these.

Reply:
These are very lightweight classes - I'd argue that the performance hit is negligible, perhaps even nonexistent.

Personally, I'd prefer to have these immediately available to plugin providers. Primarily because I don't want teams to implement these features on their own because they didn't realize that we had already done it for them - for example, the functions team implemented their own solution for `--json` because they didn't realize that we had already done it.

It's easier for plugin providers if we say, "Everything you need for plugin development is in `SfCommand`" than if we say, "Here are 5 different classes that you can use for plugin development". Especially considering, that we'll inevitably end up supporting developers who have never worked in typescript/node before.

Reply:
Fair enough. I'll run some perf tests later to see what the impact actually might be but I agree it will be minimal.